proto/asn1/asn1-universal-types/asn1_universal_types_to_der.cc

```diff
@@ -0,0 +1,73 @@
+#include "asn1_universal_types_to_der.h"
+
+namespace asn1_universal_types {
+
+std::vector<uint8_t> ASN1UniversalTypesToDER::EncodeBitString(
+    const BitString& bit_string) {
+  std::vector<uint8_t> der;
+  // BitString has tag number 3 and can be both constructed and primitive
+  // (X.208, Table 1).
+  // Add one to the length for the unused bits byte.
+  EncodeTagAndLength((bit_string.encoding() << 5 | 0x03),
+                     bit_string.val().size() + 1, 0, der);
```

Comment:
When we'd discussed this, we discussed doing this after, rather than before.

The reason being wanting to avoid magic values here (like the "+1") and instead use the length of der.